beets/importer.py

```diff
@@ -1085,7 +1122,7 @@ def read_tasks(session):
 
             log.debug(u'extracting archive {0}',
                       displayable_path(toppath))
-            archive_task = ArchiveImportTask(toppath)
+            archive_task = task_factory.archive(toppath)[0]
             try:
                 archive_task.extract()
```

Comment:
Yes, you're right.
But fixing this is a bit more complicated: We can execute every archive task, that's easy. But which path will we use for further importing? Two archive tasks may place the extracted files on different locations.
I'm not sure how to solve this. We could only allow a single archive task and stop (or continue with the next file) when more than one archive task is returned. We could also only execute the first or last archive task. Or execute every task but only use the path of the last task.